### PR TITLE
added DSF for plan in google_bigquery_capacity_commitment

### PR DESCRIPTION
**Pull request by shuyama1**
fixes https://github.com/hashicorp/terraform-provider-google/issues/14672

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by shuyama1** [comment]
Tests passed on TC: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/403610?buildTab=tests&status=passed

File: mmv1/templates/terraform/constants/bigquery_reservation_capacity_commitment.go.erb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+func bigqueryReservationCapacityCommitmentPlanDiffSuppressFunc(_, old, new string, _ *schema.ResourceData) bool {
+	if (old == "FLEX" || new == "MONTHLY" || old == "ANNUAL") && new == old+"_FLAT_RATE" {
```
Comment (roaks3) [inline mmv1/templates/terraform/constants/bigquery_reservation_capacity_commitment.go.erb:2]: Can you help me understand this line (a comment might be helpful as well)? My understanding is: we suppress the diff if the change is `FLEX -> FLEX_FLAT_RATE` or `ANNUAL -> ANNUAL_FLAT_RATE`, and that seems correct. But I think `new == MONTHLY` can never pass the second condition (`MONTHLY != ???_FLAT_RATE`).

**Review by roaks3 — COMMENTED** [review]
I had a question about the diff suppress logic, but otherwise LGTM

File: mmv1/templates/terraform/constants/bigquery_reservation_capacity_commitment.go.erb (hunk 1 above)
Comment (shuyama1) [reply mmv1/templates/terraform/constants/bigquery_reservation_capacity_commitment.go.erb:2]: oops! That's a typo. Nice catch!

**Review by roaks3 — APPROVED** [review]

**Pull request merged**